### ISSUE TITLE
Doesn't pass accessibility audit "autocomplete attribute must be used correctly"

### ISSUE THREAD

**Issue by IanLunn** [stripe/react-stripe-elements#295]
### Summary

I am using an accessibility tool called [Axe](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd) by [Deque](https://www.deque.com/). It analyses a page for accessibility concerns and offers suggestions on how to fix them.

On the [Stripe Elements examples](https://stripe.github.io/elements-examples/) page I am getting the following issues reported by Axe:

**autocomplete attribute must be used correctly (Impact: serious)**
element location: 

```
iframe[name="__privateStripeFrame7"] .InputElement
```

element source: 

```
<input type="tel" autocomplete="cc-number" autocorrect="off" spellcheck="false" name="cardnumber" class="InputElement is-empty" aria-label="Credit or debit card number" placeholder="1234 1234 1234 1234" aria-placeholder="1234 1234 1234 1234" aria-invalid="false" value="">
```

**id attribute value must be unique (Impact: minor)**

element location: 
```
iframe[name="__privateStripeFrame4"] .CardBrandIcon-inner--front > svg > g[fill="none"][fill-rule="evenodd"]
```

element source: 

```
<g id="Page-1" fill="none" fill-rule="evenodd">
```

These issues are repeated several times over for various elements. There are additional issues on the example page but I believe the two above relate to the actual Stripe Elements application rather than the example page itself.

Any plans to fix these issues?

**Comment by cweiss-stripe** [comment]
@IanLunn Thanks for letting us now about those issue.
We'll take a look at them and will let you know once they are addressed.

**Comment by asolove-stripe** [comment]
Hi there, @IanLunn, thanks for reporting this!

We've taken a deeper look at the report from this scanner tool and do not think that these are actual issues that impact accessibility or actually violate appropriate standards. That said, we are open to hear feedback if these represent actual accessibility problems.

1. `autocomplete=cc-number`. Getting the `autocomplete` attribute wrong on a credit card field would definitely be a problem! I skimmed the implementation of Axe and think the rule that's failing is [a check of which input `type`s can be used with which `autocomplete` values](https://github.com/dequelabs/axe-core/commit/e6189ce#diff-c8be89ccd7cd8049ef39d04c207728cbR44). The Axe codebase encodes that `autocomplete=cc-number` can only be used with an `type=text`, but the [WCAG standards](https://auto-wcag.github.io/auto-wcag/rules/SC1-3-5-autocomplete-valid.html) just say that the `type` should be appropriate for entering a value of the type supported by `autocomplete` without enumerating specific values for `cc-number`. I can therefore imagine why the scanner is upset at using `type=tel` to enter a card number. But this is a commonly-used method to force numeric rather than alpha keyboards in mobile browsers, which is itself a good accessibility practice for folks on mobile devices who have difficulty with precise movements. Since the field itself has a good aria label, I do not think the use of the `type=tel` represents an accessibility problem. (I tried it out in two screen readers and did not get notified of the telephone type at all, for example.) 

2. The duplicated `id` inside an svg icon does not seem to represent an accessibility problem. The svg is already marked as `role=presentation`, so nothing in there ought to be treated as anything but a description of a non-critical image.

That said, this is all my best effort after reading the standards and a bit of research with the screen readers I had at hand. If someone with more knowledge or direct experience can let us know that these are real accessibility issues, we are very happy to reopen, look further, and fix.
